Docs: Add a multi-tenant warning. (#12091)
kubernetes/ingress-nginx: docs/faq.md

# FAQ

## Multi-tenant Kubernetes

> Do not use in multi-tenant Kubernetes production installations. This project assumes that users that can create Ingress objects are administrators of the cluster

Reviewer: ```suggestion
Do not use in multi-tenant Kubernetes production installations. This project assumes that users that can create Ingress objects are administrators of the cluster.
```

> For example, the Ingress-nginx control plane has global and per ingress conifguration options that make it unsecure, if enabled, in a multitenant environment. 

Reviewer: ```suggestion
For example, the Ingress NGINX control plane has global and per Ingress configuration options that make it insecure, if enabled, in a multi-tenant environment. 
```

> For example, enabling snippets, a global configuration. Allows any ingress object to run arbitary lua code that could affect the security of all ingress objects that a controller is running. 

Reviewer: ```suggestion
For example, enabling snippets, a global configuration, allows any Ingress object to run arbitrary Lua code that could affect the security of all ingress objects that a controller is running. 
```

> For example, enabling snippets, a global configuration, allows any Ingress object to run arbitrary Lua code that could affect the security of all ingress objects that a controller is running. 

Reviewer: ```suggestion
For example, enabling snippets, a global configuration, allows any Ingress object to run arbitrary Lua code that could affect the security of all Ingress objects that a controller is running. 
```

> We changed the default to allow snippets to false in https://github.com/kubernetes/ingress-nginx/pull/10393

Reviewer: ```suggestion
We changed the default to allow snippets to `false` in https://github.com/kubernetes/ingress-nginx/pull/10393.
```
